nifcloud/acc/dhcp_option_test.go

```diff
@@ -0,0 +1,225 @@
+			}
+		}
+
+		if nifcloud.StringValue(dhcpOption.DhcpConfigurationSet[0].Key) == "domain-name-server" {
+			if nifcloud.StringValue(dhcpOption.DhcpConfigurationSet[0].ValueSet[0].Value) != "192.168.0.1" {
+				return fmt.Errorf("bad domain_name_servers state, expected \"192.168.0.1\", got: %#v", dhcpOption.DhcpConfigurationSet[0].ValueSet[0].Value)
+			}
+		}
+
+		if nifcloud.StringValue(dhcpOption.DhcpConfigurationSet[0].Key) == "domain-name-server" {
+			if nifcloud.StringValue(dhcpOption.DhcpConfigurationSet[0].ValueSet[1].Value) != "192.168.0.2" {
+				return fmt.Errorf("bad domain_name_servers state, expected \"192.168.0.2\", got: %#v", dhcpOption.DhcpConfigurationSet[0].ValueSet[1].Value)
+			}
+		}
+
+		if nifcloud.StringValue(dhcpOption.DhcpConfigurationSet[0].Key) == "ntp_servers" {
+			if nifcloud.StringValue(dhcpOption.DhcpConfigurationSet[0].ValueSet[0].Value) != "192.168.0.1" {
+				return fmt.Errorf("bad ntp_servers state, expected \"192.168.0.1\", got: %#v", dhcpOption.DhcpConfigurationSet[0].ValueSet[0].Value)
+			}
+		}
+
+		if nifcloud.StringValue(dhcpOption.DhcpConfigurationSet[0].Key) == "netbios_name_servers" {
+			if nifcloud.StringValue(dhcpOption.DhcpConfigurationSet[0].ValueSet[0].Value) != "192.168.0.1" {
+				return fmt.Errorf("bad netbios_name_servers state, expected \"192.168.0.1\", got: %#v", dhcpOption.DhcpConfigurationSet[0].ValueSet[0].Value)
+			}
+		}
+
+		if nifcloud.StringValue(dhcpOption.DhcpConfigurationSet[0].Key) == "netbios_name_servers" {
+			if nifcloud.StringValue(dhcpOption.DhcpConfigurationSet[0].ValueSet[1].Value) != "192.168.0.2" {
+				return fmt.Errorf("bad netbios_name_servers state, expected \"192.168.0.2\", got: %#v", dhcpOption.DhcpConfigurationSet[0].ValueSet[1].Value)
+			}
+		}
+
+		if nifcloud.StringValue(dhcpOption.DhcpConfigurationSet[0].Key) == "netbios_node_type" {
+			if nifcloud.StringValue(dhcpOption.DhcpConfigurationSet[0].ValueSet[0].Value) != "1" {
+				return fmt.Errorf("bad netbios_node_type state, expected \"1\", got: %#v", dhcpOption.DhcpConfigurationSet[0].ValueSet[0].Value)
+			}
+		}
+
+		if nifcloud.StringValue(dhcpOption.DhcpConfigurationSet[0].Key) == "lease_time" {
+			if nifcloud.StringValue(dhcpOption.DhcpConfigurationSet[0].ValueSet[0].Value) != "600" {
+				return fmt.Errorf("bad lease_time state, expected \"600\", got: %#v", dhcpOption.DhcpConfigurationSet[0].ValueSet[0].Value)
+			}
+		}
+		return nil
+	}
+}
+
+func testAccDhcpOptionResourceDestroy(s *terraform.State) error {
+	svc := testAccProvider.Meta().(*client.Client).Computing
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "nifcloud_dhcp_option" {
+			continue
+		}
+
+		res, err := svc.DescribeDhcpOptionsRequest(&computing.DescribeDhcpOptionsInput{
+			DhcpOptionsId: []string{rs.Primary.ID},
+		}).Send(context.Background())
+
+		if err != nil {
+			var awsErr awserr.Error
+			if errors.As(err, &awsErr) && awsErr.Code() != "Client.InvalidParameterNotFound.DhcpOption" {
```

Comment:
```suggestion
			if errors.As(err, &awsErr) && awsErr.Code() != "Client.InvalidParameterNotFound.DhcpOptionsId" {
```

DescribeDhcpOptions returns the error code `Client.InvalidParameterNotFound.DhcpOptionsId` when specified ID is not found.

```
$ nifcloud-debugcli computing describe-dhcp-options --dhcp-options-id dopt-12345678

An error occurred (Client.InvalidParameterNotFound.DhcpOptionsId) when calling the DescribeDhcpOptions operation: The dhcpOptionsId 'dopt-12345678' does not exist.
```